### PR TITLE
Python Path Mappings now provide both drive letter casings for Windows

### PR DESCRIPTION
**Pull request by awschristou**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Infrastructure (change which improves the lifecycle management (CI/CD, build, package, deploy, lint, etc) of the application, but does not change functionality.)
- [ ] Technical debt (change which improves the maintainability of the codebase, but does not change functionality)
- [ ] Testing (change which modifies or adds test coverage, but does not change functionality)
- [ ] Documentation (change which modifies or adds documentation, but does not change functionality)

## Description

When VS Code passes paths to the python debugger (on Windows), paths are passed with a capital drive letter, lowercase drive letter, or combination of the two. Notably in VS Code 1.32 to 1.33 this changed from using uppercase more often to using lowercase more often. By calling sam cli, and running within a linux based docker container, this causes path mapping mis-matches due to casing. This causes behavior like breakpoints not being defined (the debugger produces a message stating it doesn't know where to locate the particular file).

The Python debugger should ideally handle these differences when connecting with a Windows client, but for the present time, we can map both path casings, so that the debugger is able to resolve things like breakpoint locations.

See "Related Issues" for similar issues within the python debugger chain.

## Related Issue(s)

#484 

Here are some similar issues in python debuggers:

Python Extension:
https://github.com/Microsoft/vscode-python/issues/5094
https://github.com/Microsoft/vscode-python/issues/5075

ptvsd:
https://github.com/Microsoft/ptvsd/issues/1297
https://github.com/Microsoft/ptvsd/issues/1311

## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/lambda/local/debugConfiguration.ts [hunk 1]
```diff
@@ -23,13 +23,13 @@ export interface NodejsDebugConfiguration extends DebugConfiguration {
     readonly skipFiles?: string[]
 }
 
+export interface DebugConfigurationPathMapping {
```
Comment (mpiroc) [inline src/lambda/local/debugConfiguration.ts]: Each debugger has its own structure for path mappings. For example, for `coreclr` (.NET Core) it's

```typescript
{
    [remotePath: string]: /* local path */ string
}
```

Suggest to rename to something like `PythonDebugConfigurationPathMapping` or `PythonPathMapping` to clarify that it is debugger-specific.

File: src/shared/codelens/pythonCodeLensProvider.ts [hunk 2]
```diff
@@ -149,32 +151,43 @@ def ${debugHandlerFunctionName}(event, context):
     }
 }
 
-const fixFilePathCapitalization = (filePath: string): string => {
+export function getLocalRootVariants(filePath: string): string[] {
     if (process.platform === 'win32') {
-        // TODO : TEMP : https://github.com/aws/aws-toolkit-vscode/issues/484
-        return filePath.replace(/^[a-z]/, match => match.toLowerCase())
+        if (PATH_STARTS_WITH_DRIVE_LETTER_REGEX.test(filePath)) {
```
Comment (mpiroc) [inline src/shared/codelens/pythonCodeLensProvider.ts]: This condition is redundant. If it doesn't match, then `filePath.replace` will no-op.

Comment (awschristou) [reply src/shared/codelens/pythonCodeLensProvider.ts]: If the path is not a local windows path, it should be returned as-is (eg: //someshare/src/x.js )  - there are tests to validate this input too.

File: src/lambda/local/debugConfiguration.ts (hunk 1 above)
Comment (awschristou) [reply src/lambda/local/debugConfiguration.ts]: great call -- I didn't realize other runtimes had different path mapping structures

File: src/test/shared/codelens/pythonCodeLensProvider.test.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+import * as assert from 'assert'
+import { getLocalRootVariants } from '../../../shared/codelens/pythonCodeLensProvider'
+
+describe('getLocalRootVariants', async () => {
+    if (process.platform === 'win32') {
```
Comment (bryceitoc9) [inline src/test/shared/codelens/pythonCodeLensProvider.test.ts]: Will you be able to run these tests upstream on a build box?

Comment (awschristou) [reply src/test/shared/codelens/pythonCodeLensProvider.test.ts]: We already have some platform specific tests that run on mac (Travis) and linux (Travis, CodeBuild)

Comment (bryceitoc9) [reply src/test/shared/codelens/pythonCodeLensProvider.test.ts]: Confirmed that Travis doesn't run these tests (see: https://travis-ci.org/aws/aws-toolkit-vscode/jobs/521971775#L1062 ), we may add Windows tests at a later point: https://blog.travis-ci.com/2018-10-11-windows-early-release

**Pull request merged**